### ISSUE TITLE
Crash when throws keyword is added to swift functions used in kotlin

### ISSUE THREAD

**Issue by Ti-m** [JetBrains/kotlin-native#3825]
Hey,

I stumbled about this in the docs:

„Note that the opposite reversed translation is not implemented yet: Swift/Objective-C error-throwing methods aren't imported to Kotlin as exception-throwing.“
https://kotlinlang.org/docs/reference/native/objc_interop.html#errors-and-exceptions

I tried this, and it is not simply ignored, instead it crashes every time, if the function has a closure parameter:
`func functionWithCallback(callback: @escaping (KotlinInt, String) -> Void) throws {} //This throws produces a crash Thread 1: EXC_BAD_ACCESS (code=1, address=0x10)`
If the function has no parameter it works:
`func function() throws {} //this works`

I call it like:
```
let someKotlinInterfaceImplementationInSwift = SomeInterfaceImplementationInSwift()
let someKotlinClass = SomeKotlinClass(dependencieImplementedInSwift: someKotlinInterfaceImplementationInSwift)
someKotlinClass.someFunctionImplementedInKotlin()
```
 ```
Thread 1 Queue : com.apple.main-thread (serial)
 #0    0x0000000106c215ff in thunk for @escaping @callee_unowned @convention(block) (@unowned GreetingInt, @unowned NSString) -> () ()
 #1    0x0000000106c213bc in SomeInterfaceImplementationInSwift.functionWithCallback(callback:) at /Users/timb/projects/mpp-example-exceptions/iosApp/iosApp/SomeInterfaceImplementation.swift:13
 #2    0x0000000106c214c5 in @objc SomeInterfaceImplementationInSwift.functionWithCallback(callback:) ()
 #3    0x0000000106f9e7d8 in ___lldb_unnamed_symbol3$$greeting ()
 #4    0x0000000106f992be in kfun:org.greeting.SomeKotlinClass.someFunctionImplementedInKotlin() at /Users/timb/projects/mpp-example-exceptions/greeting/src/commonMain/kotlin/SomeKotlinClass.kt:6
 #5    0x0000000106f9e6dd in objc2kotlin.65 at /Users/timb/projects/mpp-example-exceptions/greeting/src/iosMain/kotlin/<compiler-generated>:1
 #6    0x0000000106c20510 in ViewController.viewDidLoad() at /Users/timb/projects/mpp-example-exceptions/iosApp/iosApp/ViewController.swift:21
```

 I made a simple reproducer: https://github.com/Ti-m/mpp-example/tree/problem-with-exceptions

**Comment by SvyatoslavScherbina** [comment]
Thank you for the report.

I'm afraid this crash has nothing to do with Kotlin and likely caused by a bug in Swift involving Objective-C interoperability.

Reproducer:

Swift:
```swift
import Foundation

class CallerImpl : Caller {
    func call(_ callback: (() -> Void)?) throws {
        print("call")
        print(callback ?? "fail")
        callback!()
    }
}

call(CallerImpl())
```

Objective-C:
```obj-c
#import <Foundation/Foundation.h>

@protocol Caller
@required
- (BOOL)call:(void (^)(void))callback error:(NSError**)error;
@end;

void call(id<Caller> caller) {
    NSError* error = nil;
    void (^block)(void) = ^void() { NSLog(@"OK"); };
    NSLog(@"%@", block);
    [caller call:block error:&error];
}
```

**Comment by Ti-m** [comment]
Ok thx anyway for clarification.

**Comment by SvyatoslavScherbina** [comment]
Somewhat fixed with #3852.
